client/signup/steps/woocommerce-install/hooks/use-eligibility/index.ts

```diff
@@ -0,0 +1,121 @@
+
+const TRANSFERRING_NOT_BLOCKERS = [
+	eligibilityHoldsConstants.NO_BUSINESS_PLAN, // let's redirect to checkout page
+	eligibilityHoldsConstants.TRANSFER_ALREADY_EXISTS, // let's handle it in transferring endpoint.
+];
+
+type EligibilityHook = {
+	eligibilityHolds?: string[];
+	eligibilityWarnings?: EligibilityWarning[];
+	isFetching: boolean;
+	wpcomDomain: string | null;
+	stagingDomain: string | null;
+	pluginsWarning: EligibilityWarning[];
+	widgetsWarning: EligibilityWarning[];
+	wpcomSubdomainWarning: EligibilityWarning | undefined;
+	transferringBlockers: string[];
+	hasBlockers: boolean;
+	siteNeedUpgrade: string | false;
+};
+
+export default function useEligibility( siteId: number ): EligibilityHook {
+	const dispatch = useDispatch();
+
+	// Request eligibility data.
+	useEffect( () => {
+		if ( ! siteId ) {
+			return;
+		}
+
+		dispatch( fetchAutomatedTransferStatusOnce( siteId ) );
+		dispatch( requestEligibility( siteId ) );
+	}, [ siteId, dispatch ] );
+
+	// Get eligibility data.
+	const { eligibilityHolds, eligibilityWarnings }: EligibilityData = useSelector( ( state ) =>
+		getEligibility( state, siteId )
+	);
+
+	// Pick the wpcom subdomain.
+	const wpcomDomain = useSelector( ( state ) => getSiteDomain( state, siteId ) );
+
+	// Get staging sudomain based on the wpcom subdomain.
+	const stagingDomain = wpcomDomain?.replace( /\b\.wordpress\.com/, '.wpcomstaging.com' ) || null;
+
+	// Check whether it's requesting eligibility data.
+	const isFetching = !! useSelector( ( state ) =>
+		isFetchingAutomatedTransferStatus( state, siteId )
+	);
+
+	// Check whether the wpcom.com subdomain warning is present.
+	const wpcomSubdomainWarning = eligibilityWarnings?.find(
+		( { id } ) => id === 'wordpress_subdomain'
+	);
+
+	// Plugins warnings
+	const pluginsWarning = eligibilityWarnings?.filter( ( { type } ) => type === 'plugins' ) || [];
+
+	// Widgets warnings
+	const widgetsWarning = eligibilityWarnings?.filter( ( { type } ) => type === 'widgets' ) || [];
+
+	// Trasferring blockers
+	const transferringBlockers =
+		eligibilityHolds?.filter( ( hold ) => ! TRANSFERRING_NOT_BLOCKERS.includes( hold ) ) || [];
+
+	/*
+	 * Plan site and `woop` site feature.
+	 * If the eligibility holds contains `NO_BUSINESS_PLAN`,
+	 * if the site doesn't have the `woop` feature,
+	 * and if the site has the `woop` feature upgradaable,
+	 * then the site needs to be upgraded.
+	 */
+	const eligibilityNoProperPlan = eligibilityHolds?.includes(
+		eligibilityHoldsConstants.NO_BUSINESS_PLAN
+	);
+	const isWoopFeatureActive = useSelector( ( state ) =>
+		hasActiveSiteFeature( state, siteId, FEATURE_WOOP )
+	);
+	const hasWoopFeatureAvailable = useSelector( ( state ) =>
+		hasAvailableSiteFeature( state, siteId, FEATURE_WOOP )
+	);
+
+	const siteNeedUpgrade = ! (
+		eligibilityNoProperPlan &&
+		! isWoopFeatureActive &&
+		hasWoopFeatureAvailable
+	)
+		? false
+		: `/woocommerce-installation/${ wpcomDomain }`;
```

Comment:
If we redirect to the landing page, how/when does the site get transferred to Atomic? The /woocommerce-installation flow doesn't transfer sites.

Reply:
Are you sure? I thought once the site gets the proper plan, you will be able to run the store installation which includes the site transferring process, even using the current process.

Reply:
https://user-images.githubusercontent.com/77539/144036926-bd94ba5d-a1c7-494d-9d4b-782e95d7bcaa.mov